tests/test_config.py

```diff
@@ -0,0 +1,116 @@
+from _pytest.monkeypatch import monkeypatch
+import pytest
+
+from dpytools.config.config import Config
+from dpytools.config.properties.string import StringProperty
+from dpytools.config.properties.intproperty import IntegerProperty
+
+def test_config_loader(monkeypatch):
+    """
+    Tests that a config object can be created and its attributes 
+    dynamically generated from an input config dictionary with the 
+    expected contents.
+    """
+
+    # Assigning environment variable values for config dictionary values
+    monkeypatch.setenv("SOME_STRING_ENV_VAR", "Some string value")
+    monkeypatch.setenv("SOME_URL_ENV_VAR", "https://test.com/some-url")
+    monkeypatch.setenv("SOME_INT_ENV_VAR", "6")
+
+    config_dictionary = {
+    "SOME_STRING_ENV_VAR": {
+        "class": StringProperty,
+        "property": "name1",
+        "kwargs": {
+            "regex": "I match a thing",
+            "min_len": 10
+        },
+    },
+    "SOME_URL_ENV_VAR": {
+        "class": StringProperty,
+        "property": "name2",
+        "kwargs": {
+            "regex": "https://.*",
+            "max_len": 100
+        },
+    },
+    "SOME_INT_ENV_VAR": {
+        "class": IntegerProperty,
+        "property": "name3",
+        "kwargs": {
+            "min_val": 5,
+            "max_val": 27
+        }
+    },
+}
+
+    config = Config.from_env(config_dictionary)
+    
+    # Assertions
+
+    assert config.name1._name == "name1"
```

Comment:
can you use just `name1.name` and `name1.value` for these please. Should work once you've addressed the other comment on properties. That way we're also checking the get/set property methods are working.

The idea is we expect/advertise (via the docs) those non `_` methods only, so we get that protection against people doing silly things (or at least they'll have to go out of their way to, which is the best we can hope for).